src/s390x/zipl.rs

```diff
@@ -0,0 +1,191 @@
+    let mut contents = read_options(&boot_config)?;
+
+    let pat = "options ";
+    let idx = contents
+        .find(pat)
+        .chain_err(|| format!("cannot find: \'{}\'", pat))?
+        + pat.len();
+
+    let s390_options = get_s390x_kargs()?;
+    contents.insert_str(idx, &s390_options);
+
+    eprintln!("Writing boot options");
+    std::fs::write(&boot_config, &contents)
+        .chain_err(|| format!("writing {}", boot_config.display()))?;
+
+    eprintln!("Generating zipl config");
+    let mut contents = &contents[idx..contents.len()];
+    if let Some(idx) = contents.find('\n') {
+        contents = &contents[0..idx];
+    }
+
+    zipl_config
+        .write_all(contents.as_bytes())
+        .chain_err(|| "writing zipl config")?;
+
+    if let Some(kargs) = firstboot_kargs.as_ref() {
+        zipl_config
+            .write_all(kargs.as_bytes())
+            .chain_err(|| "writing zipl config")?;
+    }
+
+    zipl_config
+        .write_all(b" ignition.firstboot")
+        .chain_err(|| "writing zipl config")?;
+
+    Ok(())
+}
+
+fn find_file<P: AsRef<Path>>(root: P, pat: &str) -> Result<PathBuf> {
+    let mut path = PathBuf::new();
+    path.push(&root);
+    for entry in WalkDir::new(&path).into_iter().filter_map(|e| e.ok()) {
+        if !entry.file_type().is_file() {
+            continue;
+        }
+        let name = entry.file_name().to_string_lossy();
+        if pat.starts_with('.') {
+            if !name.ends_with(pat) {
+                continue;
+            }
+        } else if !name.starts_with(pat) {
+            continue;
+        }
+
+        return Ok(entry.path().to_path_buf());
+    }
+    Err(format!(
+        "Cannot find file with mask: {} in {}",
+        pat,
+        root.as_ref().display()
+    )
+    .into())
+}
+
+fn boot_config<P: AsRef<Path>>(boot: P) -> Result<PathBuf> {
+    let mut path = PathBuf::new();
+    path.push(boot);
+    path.push("loader");
+    path.push("entries");
+    find_file(&path, ".conf")
+}
+
+fn kernel<P: AsRef<Path>>(boot: P) -> Result<PathBuf> {
+    let mut path = PathBuf::new();
+    path.push(boot);
+    path.push("ostree");
+    find_file(&path, "vmlinuz")
+}
+
+fn initramfs<P: AsRef<Path>>(boot: P) -> Result<PathBuf> {
+    let mut path = PathBuf::new();
+    path.push(boot);
+    path.push("ostree");
+    find_file(&path, "initram")
+}
+
+fn read_options<P: AsRef<Path>>(filename: P) -> Result<String> {
+    let mut contents = std::fs::read_to_string(&filename)
+        .chain_err(|| format!("reading {}", filename.as_ref().display()))?;
+    contents.retain(|c| c != '\n' && c != '\r');
+    contents.push_str(" ");
+    Ok(contents)
+}
+
+fn get_s390x_kargs() -> Result<String> {
+    let re = Regex::new(
+        r#"rd.dasd=[[[:alnum:]].]*|rd.zfcp=[[[:alnum:]].,=:!]*|rd.znet=[[[:alnum:]].,=]*|zfcp.allow_lun_scan=[\d]|cio_ignore=[[[:alnum:]].,=:!]*"#)
+        .chain_err(|| "compiling regex")?;
+    let cmdline = read_options("/proc/cmdline")?;
```

Comment:
In general, coreos-installer doesn't rely on external state like this.  Instead, it takes configuration explicitly from its own command line.  We can do otherwise if there's a good reason, though.  What's the purpose of these kernel arguments?  Do we always want to use exactly the ones from the current boot?

Reply:
@nikita-dubrovskii: I think some thing like this : https://github.com/coreos/coreos-installer/blob/ce8e9a8ccbf4fb0b5703d367edea5662e3137f8a/dracut/30coreos-installer/parse-coreos.sh#L77 instead of `/proc/cmdline`.

Reply:
And for the coreos-installer bin itself : https://github.com/coreos/coreos-installer/blob/ce8e9a8ccbf4fb0b5703d367edea5662e3137f8a/coreos-installer#L702

Reply:
files in /tmp are no longer available

Reply:
another way is parse those kcmdline you want in /proc/cmdline in `systemd/coreos-installer-generator`

Reply:
`systemd/coreos-installer-service` : so that you can make the s390x parameters in /proc/cmdline you want as an option for `coreos-installer`. Like ``--dasd-id 0123`.

Reply:
why should i? don't see any benefits in doing this. main question - do we need all s390x specific + networking (firstboot) args at all? for some reasons legacy was using all of them. @tuan-hoang1 why it was like that in legacy?

Reply:
i've checked CoreOS on zVM without s390x specific args - it fails during boot. But firstboot args could be ignored